src/main.cpp

```diff
@@ -2287,11 +2293,12 @@ bool Session::InitializeDRM()
       return false;
     }
 
-    if (!decrypter_->OpenDRMSystem(license_key_.c_str(), server_certificate_, drmConfig_))
-    {
-      kodi::Log(ADDON_LOG_ERROR, "OpenDRMSystem failed");
-      return false;
-    }
+    if (!decrypter_->HasCdmSession())
+      if (!decrypter_->OpenDRMSystem(license_key_.c_str(), server_certificate_, drmConfig_))
+      {
+        kodi::Log(ADDON_LOG_ERROR, "OpenDRMSystem failed");
+        return false;
+      }
```

Comment:
If you are using an enclosed if statement always use brackets for the outer one. Then you can't run foul of a dangling else when extending the code.

Reply:
No worries, sorted!